### PR TITLE
Disable attachment carousel scroll when zoomed in

### PR DESCRIPTION
**Pull request by terrysahaidak**
### Details

There are many problems with the current image zoom component. First of all, it's deprecated by the authors. There are workarounds for double ckick for zoom, it doesn't zoom when you pinch until you double press etc, but also it uses PanResponder and runs all the logic on the JS thread, so when JS thread is blocked by something it doesn't respond to gestures. 

And there was another bug - if you zoom in, the pager's scroll view would scroll alongside the pan responder.
Since the whole component is deprecated and we already have quite a few workarounds I decided to just block the scroll completely when you zoom in, since on iOS it doesn't work properly, on Android it sometimes might work, but there is no clear UX pattern when it works. 

There is no easy way to fix this properly so it works on both platforms well.

I already tried:
- Using react-native-pager-view and changing `scrollEnabled` using `react-native-reanimated`'s `useAnimatedProps`, but it still doesn't allow for the proper swipe to the left/right to change the page
- Using `react-native-gesture-handler` and scroll view where we would block the gesture until we `enable` it on the UI thread, but it requires the zoom component to be written with reanimated and gesture handler as well, but also this same behavior doesn't properly work on Android, i might require some fixes in the gesture handler code

So in order to properly fix this component for good, we need to rewrite it with gesture-handler and reanimated, so all the logic is running on UI thread, then we can combine gesture-handlers for single tap, double tap, pan and pinch gestures in order to enable rich UX for the image zoom.

But in the meantime this change allows users to actually use the app with a slight inconvenience - they have to zoom out to navigate to the next page.

### Fixed Issues

$ https://github.com/Expensify/App/issues/17720
PROPOSAL: GH_LINK_ISSUE(COMMENT)

### Tests

- Open chat with multiple image attachments
- Open image
- Zoom in by double tap
- Swipe left/right
- It should not switch the page until you zoom out by double tap/pinch out
- [x] Verify that no errors appear in the JS console

### QA Steps

- Open chat with multiple image attachments
- Open image
- Zoom in by double tap
- Swipe left/right
- It should not switch the page until you zoom out by double tap/pinch out
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

No changes

</details>

<details>
<summary>Mobile Web - Chrome</summary>

No changes

</details>

<details>
<summary>Mobile Web - Safari</summary>

No changes

</details>

<details>
<summary>Desktop</summary>

No changes

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/7809008/bafd8de7-6fe2-4c83-bacb-96e60988f067

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/7809008/734ca949-8409-4098-87aa-1c2398bfd4eb

</details>

### CODE REVIEW

File: src/components/ImageView/index.native.js [hunk 1]
```diff
@@ -198,12 +201,15 @@ class ImageView extends PureComponent {
                                     scale: 2,
                                     duration: 100,
                                 });
+
+                                this.props.onScaleChanged(2);
```
Comment (terrysahaidak) [inline src/components/ImageView/index.native.js:205]: `onMove` will be called after the zoom animation, so it's possible to zoom and swipe and stuck in between the images. Sending scale just when we actually trigger the animation makes this nearly impossible, you should be really fast to catch in between state updates. And this lucky case will be fixed by migration to UI thread only code with gesture handler and reanimated.

**Comment by terrysahaidak** [comment]
@aimane-chnaif thanks a lot. this implementation was discussed basically here in this thread on Slack: https://expensify.slack.com/archives/C035J5C9FAP/p1682027016020719

The current plan we agreed on is to merge the fix to unblock the zoom feature for the users and work on a better implementation altogether.

**Review by puneetlath — DISMISSED** [review]
Seems to make sense to me.

**Comment by puneetlath** [comment]
@aimane-chnaif bump.

**Comment by puneetlath** [comment]
@aimane-chnaif bump!

**Comment by terrysahaidak** [comment]
@aimane-chnaif the bug you described is not related to the changes in this PR and is related to the image-zoom library. PDF uses a different zoom implementation this is why it doesn't happen there.

**Review by aimane-chnaif — DISMISSED** [review]
All good!
Just waiting for author checklist to be completed

**Comment by terrysahaidak** [comment]
hey, sorry for the delay, I saw in another merged PR where it also was changing something for iOS/android only and they did not check other platforms since there were no changes and provided screen recordings.

I checked everything.

I'll add the comment in the code as requested in ~2h.

**Review by aimane-chnaif — DISMISSED** [review]

**Comment by puneetlath** [comment]
@terrysahaidak you have a lint issue.

**Review by puneetlath — APPROVED** [review]

**Review by aimane-chnaif — APPROVED** [review]

**Comment by terrysahaidak** [comment]
Edited code using github.dev and as always there are some random eslint/prettier errors :(
Such fixes are so easy to do using it, i wish we would have prettier and stuff also working in there.

**Comment by puneetlath** [comment]
Oh, I've never used it. How could we get prettier working in there?

**Pull request merged**